apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nav-sidebar/src/components/nav-sidebar/index.tsx

```diff
@@ -280,20 +256,33 @@ function useNavItems(): Post[] {
 				status: statusFilter,
 			} ) || [];
 
-		const currentPost = {
+		const current = {
 			id: currentPostId,
 			slug: getEditedPostAttribute( 'slug' ),
 			status: isEditedPostNew() ? 'draft' : getEditedPostAttribute( 'status' ),
 			title: { raw: getEditedPostAttribute( 'title' ), rendered: '' },
 		};
 
-		return [ currentPost, ...items ] as any;
+		const partitionDraftPosts: Record< string, Array< Post[] > > = items.reduce(
```

Comment:
You might be able to do:

```typescript
const [ drafts, recent ] = _.partition( items, { status: 'draft' } );
return { current: [ current ], drafts, recent };
```

Disclaimer: have not tested or even checked if it is valid TS.

Reply:
I like this approach!

The [lodash function itself](https://github.com/lodash/lodash/blob/master/partition.js) is neat enough to port:

```
     const [ drafts, recent ] = items.reduce(
			( result: Post[][], item: Post ) => (
				result[ item.status === 'draft' ? 0 : 1 ].push( item ), result
			),
			[ [], [] ]
		);

		return { current: [ current ], drafts, recent } as any;
```

But since we are importing `reduce` anyway `partition` looks like it has a small enough footprint.